pkg/system/phase4_configuring.go

```diff
@@ -537,16 +541,156 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+	log := r.Logger.WithField("func", "ReconcileHPAEndpoint")
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
+	var err error
+	prometheusList := deployedPrometheus(log)
+	switch r.NooBaa.Spec.AutoscalingType {
+	case "keda":
+		if err = r.deleteExistingAutoscaler(string(r.NooBaa.Spec.AutoscalingType)); err != nil {
+			return err
+		}
+		if err = r.autoscaleKeda(prometheusList); err != nil {
+			return err
+		}
+	case "hpav2":
+		//TODO
+		log.Infof("HPAV2 implementation pending")
+	case "hpav1":
+		if err = r.deleteExistingAutoscaler(string(r.NooBaa.Spec.AutoscalingType)); err != nil {
+			return err
+		}
+		if err = r.ReconcileObject(r.HPAEndpoint, r.SetDesiredHPAEndpoint); err != nil {
+			return err
+		}
+	default:
+		log.Infof("❌ Autoscaling type is not defined for noobaa")
+		return errors.New("Autoscaling type is not defined")
+	}
+	return r.updateNoobaaEndpoint()
+
+}
+
+func (r *Reconciler) deleteExistingAutoscaler(autoscalingType string) error {
+	autoscalers := &autoscalingv1.HorizontalPodAutoscalerList{}
```

Comment:
notice that here you only list HPAV1 resources. these are not relevant for KEDA for example. 
I suggest having a dedicated cleanup function for each type of autoscaler and from here call the relevant cleanups according to `autoscalingType`. e.g.:

```go
if autoscalingType == `nbv1.AutoscalingTypeKeda` {
   cleanupHPAV1()
   cleanupHPAV2()
}
if autoscalingType == `nbv1.AutoscalingTypeHPAV1` {
   cleanupKEDA()
   cleanupHPAV2()
}
// ... etc.
```